static/app/views/settings/account/mergeAccounts.tsx

```diff
@@ -0,0 +1,343 @@
+        />
+        <div>
+          <Button
+            priority="danger"
+            onClick={() => handleSubmit(selectedUserIds, tokenValue)}
+          >
+            Submit
+          </Button>
+        </div>
+      </List>
+    </Fragment>
+  );
+}
+
+export default MergeAccounts;
+
+function makeMergeAccountsEndpointKey(): ApiQueryKey {
+  return [ENDPOINT];
+}
+
+function RenderSelectAccounts({users, onSelect}: RenderSelectAccountsProps) {
+  const signedInUser = useUser();
+
+  const currentAccount = users.filter(({id}) => id === signedInUser.id);
+  const otherAccounts = users.filter(({id}) => id !== signedInUser.id);
+
+  return (
+    <Fragment>
+      <StyledListItem>{t('Select Your Accounts')}</StyledListItem>
+      <TextBlock>
+        {tct(
+          `Select the accounts that you want to merge into your currently active account,
+          then confirm and merge. [strong:The accounts that you do not select will be deleted!]`,
+          {
+            strong: <strong />,
+          }
+        )}
+      </TextBlock>
+      <TextBlock>{t(`Your currently active account:`)}</TextBlock>
+      <UserContext value>
+        <Users users={currentAccount} onSelect={onSelect} />
+      </UserContext>
+      <TextBlock>{t(`Your other accounts:`)}</TextBlock>
+      <UserContext value={false}>
+        <Users users={otherAccounts} onSelect={onSelect} />
+      </UserContext>
+    </Fragment>
+  );
+}
+
+type RenderSelectAccountsProps = {
+  onSelect: (newUserId: string) => void;
+  users: UserWithOrganizations[];
+};
+
+type UserRowProps = {
+  id: string;
+  lastSeen: string;
+  name: string;
+  onSelect: (newUserId: string) => void;
+  organizations: string;
+};
+
+type UserProps = {
+  onSelect: (newUserId: string) => void;
+  users: UserWithOrganizations[];
+};
+
+function Users({users, onSelect}: UserProps) {
+  const isPrimaryUser = useContext(UserContext);
+  if (isPrimaryUser) {
+    return (
+      <Panel>
+        <UserPanelHeader>
+          <div>{t('Name')}</div>
+          <div>{t('Last Active')}</div>
+          <div>{t('Organizations')}</div>
+        </UserPanelHeader>
+        <PanelBody>
+          {users.map(userObj => (
+            <UserRow
+              onSelect={onSelect}
+              key={userObj.id}
+              id={userObj.id}
+              name={userObj.name}
+              lastSeen={userObj.lastActive}
+              organizations={userObj.organizations.join(', ')}
+            />
+          ))}
+        </PanelBody>
+      </Panel>
+    );
+  }
+  return (
+    <Panel>
+      <UserPanelHeader>
+        <div>{t('Name')}</div>
+        <div>{t('Last Active')}</div>
+        <div>{t('Organizations')}</div>
+        <div>{t('Merge')}</div>
+      </UserPanelHeader>
+      <PanelBody>
+        {users.map(userObj => (
+          <UserRow
+            onSelect={onSelect}
+            key={userObj.id}
+            id={userObj.id}
+            name={userObj.name}
+            lastSeen={userObj.lastActive}
```

Comment:
instead of passing each individual property of the user, maybe pass the entire thing